### PR TITLE
For #17917: Use View binding in tracking protection

### PR DESCRIPTION
**Pull request by codrut-topliceanu**
For https://github.com/mozilla-mobile/fenix/issues/17917

### Pull Request checklist

- [x] **Tests**: This PR includes no tests as it is only a minor change
- [x] **Screenshots**: This PR does not include screenshots as no user visible changes have been made.
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md).

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by codrut-topliceanu** [comment]
image
...
image


trash-computer

**Comment by Amejia481** [comment]
Please let's 🙏🏽  hold any major work related to the lock and tracking protection dialogs, until we land  https://github.com/mozilla-mobile/fenix/pull/20536

**Comment by jonalmeida** [comment]
> We disabled some intermittent tests hope it helps #20798

And I'm re-enabled them in https://github.com/mozilla-mobile/fenix/pull/20805 ! 😄

File: app/src/main/java/org/mozilla/fenix/trackingprotection/TrackingProtectionOverlay.kt
Comment (Mugurell) [inline app/src/main/java/org/mozilla/fenix/trackingprotection/TrackingProtectionOverlay.kt:175]: One small thing to remove.

**Review by Mugurell — APPROVED** [review]
LGTM with a big asterisk for the `kotlinx.android.extensions.LayoutContainer` still used in `ExceptionsView` that is to be removed in https://github.com/mozilla-mobile/fenix/pull/20813

File: app/src/main/java/org/mozilla/fenix/trackingprotection/TrackingProtectionOverlay.kt
Comment (codrut-topliceanu) [reply app/src/main/java/org/mozilla/fenix/trackingprotection/TrackingProtectionOverlay.kt:175]: oops

**Comment by Mugurell** [comment]
Pending tracking protection work. Will hold off my review.

**Comment by Amejia481** [comment]
The PR for merging the tracking protection and lock icon has landed https://github.com/mozilla-mobile/fenix/pull/20536, please proceed :)

**Comment by codrut-topliceanu** [comment]
Too many conflicts. Recreated PR in a different branch, see : https://github.com/mozilla-mobile/fenix/pull/20886

**Pull request closed without merging**
